code/modules/power/turbine/turbine.dm

```diff
@@ -0,0 +1,557 @@
+	var/damage = 0
+	///Used to calculate the max damage received per tick and if the alarm should be called
+	var/damage_archived = 0
+
+	///Our internal radio
+	var/obj/item/radio/radio
+	///The key our internal radio uses
+	var/radio_key = /obj/item/encryptionkey/headset_eng
+	///The engineering channel
+	var/engineering_channel = "Engineering"
+
+	COOLDOWN_DECLARE(turbine_damage_alert)
+
+/obj/machinery/power/turbine/core_rotor/Initialize(mapload)
+	. = ..()
+	radio = new(src)
+	radio.keyslot = new radio_key
+	radio.set_listening(FALSE)
+	radio.recalculateChannels()
+
+/obj/machinery/power/turbine/core_rotor/LateInitialize()
+	. = ..()
+	activate_parts()
+
+/obj/machinery/power/turbine/core_rotor/Destroy(mob/user)
+	deactivate_parts(user)
+	QDEL_NULL(radio)
+	return ..()
+
+/obj/machinery/power/turbine/core_rotor/enable_parts(mob/user)
+	. = ..()
+	if(was_complete)
+		was_complete = FALSE
+		activate_parts(user)
+
+/obj/machinery/power/turbine/core_rotor/disable_parts(mob/user)
+	. = ..()
+	if(all_parts_connected)
+		was_complete = TRUE
+	deactivate_parts()
+
+/obj/machinery/power/turbine/core_rotor/multitool_act(mob/living/user, obj/item/tool)
+	if(!all_parts_connected && activate_parts(user))
+		balloon_alert(user, "all parts are linked")
+	return TOOL_ACT_TOOLTYPE_SUCCESS
+
+/obj/machinery/power/turbine/core_rotor/multitool_act_secondary(mob/living/user, obj/item/tool)
+	if(!all_parts_connected)
+		return TOOL_ACT_TOOLTYPE_SUCCESS
+	var/obj/item/multitool/multitool = tool
+	multitool.buffer = src
+	to_chat(user, span_notice("You store linkage information in [tool]'s buffer."))
+	return TOOL_ACT_TOOLTYPE_SUCCESS
+
+/obj/machinery/power/turbine/core_rotor/proc/activate_parts(mob/user)
+
+	compressor = locate(/obj/machinery/power/turbine/inlet_compressor) in get_step(src, turn(dir, 180))
+	turbine = locate(/obj/machinery/power/turbine/turbine_outlet) in get_step(src, dir)
+
+	var/parts_present = TRUE
+	if(!compressor || !turbine)
+		if(user)
+			balloon_alert(user, "missing parts detected")
+		parts_present = FALSE
+	if(compressor.dir != dir || !compressor.can_connect)
+		if(user)
+			balloon_alert(user, "wrong compressor direction")
+		parts_present = FALSE
+	if(turbine.dir != dir || !turbine.can_connect)
+		if(user)
+			balloon_alert(user, "wrong turbine direction")
+		parts_present = FALSE
+
+	if(!parts_present)
+		all_parts_connected = FALSE
+		return FALSE
+
+	compressor.core = src
+	turbine.core = src
+
+	input_turf = get_step(compressor.loc, turn(dir, 180))
+	output_turf = get_step(turbine.loc, dir)
+
+	compressor_mixture = new
+	rotor_mixture = new
+	turbine_mixture = new
+	compressor_mixture.volume = compressor.gas_theoretical_volume
+	rotor_mixture.volume = gas_theoretical_volume
+	turbine_mixture.volume = turbine.gas_theoretical_volume
+
+	compressor_part_efficiency = compressor.installed_part.part_efficiency
+	stator_part_efficiency = turbine.installed_part.part_efficiency
+	rotor_part_efficiency = installed_part.part_efficiency
```

Comment:
Why store these variables on this machine?

Reply:
Easier access while doing stuff I can move them around to the various parts